openfermioncirq/gates/fermionic_simulation.py

```diff
@@ -86,11 +88,99 @@ def state_swap_eigen_component(x: str, y: str, sign: int = 1, angle: float=0):
     return component
 
 
+@cirq.value_equality(approximate=True)
+class FermionicSimulationGate(cirq.EigenGate):
+    r"""The Jordan-Wigner transform of :math:`\exp(-i H)` for a fermionic Hamiltonian
+    :math:`H`.
+
+    Each subclass corresponds to a set of generators :math:`\{G_i\}`
+    corresponding to the family of Hamiltonians :math:`\sum_i w_i G_i +
+    \text{h.c.}`, where the weights :math:`w_i \in \mathbb C` are specified by
+    the instance.
+
+    The Jordan-Wigner mapping maps the fermionic modes :math:`(0, \ldots, n - 1)` to qubits 
+    :math:`(0, \ldots, n - 1)`, respectively.
+
+    Each generator :math:`G_i` must be a linear combination of fermionic
+    monomials consisting of an even number of creation/annihilation operators.
+    This is so that the Jordan-Wigner transform acts only on the gate's qubits,
+    even when the fermionic modes are offset as part of a larger Jordan-Wigner string.
+    """
+
+    def __init__(self,
+                 weights: Optional[Tuple[complex, ...]]=None,
+                 absorb_exponent: bool = False,
+                 **kwargs) -> None:
+        """A fermionic interaction.
+
+        Args:
+            weights: The weights of the terms in the Hamiltonian.
+            absorb_exponent: Whether to absorb the given exponent into the
+                weights. If true, the exponent of the return gate is `1`.
+                Defaults to `False`.
+        """
+        if weights is None:
+            weights = (1.,) * self.num_weights
+        self.weights = weights
+
+        super().__init__(**kwargs)
+
+        if absorb_exponent:
+            self.absorb_exponent_into_weights()
+
+    @abc.abstractproperty
+    def num_weights(self) -> int:
+        """The number of parameters (weights) in the generator."""
+
+    @abc.abstractproperty
+    def qubit_generator(self) -> np.ndarray:
+        """The matrix G such that the gate's unitary is exp(-i t G) with
+        exponent t."""
+
+    @abc.abstractproperty
+    def fermion_generators(self) -> Iterable[openfermion.FermionOperator]:
```

Comment:
This name is confusing because there is something else called `fermion_generator`. We need to rename it. I suggest `fermion_generator_terms` or `fermion_generator_components`.

Reply:
Done (`fermion_generator_components`).